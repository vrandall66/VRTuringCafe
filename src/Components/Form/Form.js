import React, { Component } from "react";
import "./Form.css";

class Form extends Component {
  constructor() {
    super();
    this.state = {
      name: "",
      date: "",
      time: "",
      number: ""
    };
  }

  handleChange = event => {
    if (event.target.name === "number") {
      parseInt(event.target.value);
    }
    this.setState({ [event.target.name]: event.target.value });
  };

  handleSubmit = event => {
    const newReservation = { ...this.state, id: Date.now() };
    return this.props.newReservation(newReservation);
  };

  render() {
    return (
      <form className="Form>">
        <input
          type="text"
          name="name"
          id="Form__input--name"
          placeholder="Name"
          value={this.state.name}
          onChange={this.handleChange}
        />
        <input
          type="text"
          name="date"
          id="Form__input--date"
          placeholder="Date (mm/dd)"
          value={this.state.date}
          onChange={this.handleChange}
        />
        <input
          type="text"
          name="time"
          id="Form__input--time"
          placeholder="Time"
          value={this.state.time}
          onChange={this.handleChange}
        />
        <input
          type="text"
          name="number"
          id="Form__input--number"
          placeholder="Number of Guests"
          value={this.state.number}
          onChange={this.handleChange}
        />
        <button type="button" name="submit" id="Form__btn--submit" onClick={this.handleSubmit}>
          Make Reservation
        </button>
      </form>
    );
  }
}

export default Form;
