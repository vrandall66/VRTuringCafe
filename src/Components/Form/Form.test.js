import React from "react";
import { shallow } from "enzyme";
import Form from "./Form";

describe("Form", () => {
  let wrapper;
  let newReservation = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Form newReservation={newReservation} />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should update state", () => {
    const mockEvent1 = {
      target: {
        name: "name",
        value: "Robbie"
      }
    };
    const mockEvent2 = {
      target: {
        name: "date",
        value: "10/17"
      }
    };
    const mockEvent3 = {
      target: {
        name: "time",
        value: "12:00PM"
      }
    };
    const mockEvent4 = {
      target: {
        name: "number",
        value: 2
      }
    };

    const expected = {
      name: "Robbie",
      date: "10/17",
      time: "12:00PM",
      number: 2
    };

    wrapper.instance().handleChange(mockEvent1);
    wrapper.instance().handleChange(mockEvent2);
    wrapper.instance().handleChange(mockEvent3);
    wrapper.instance().handleChange(mockEvent4);

    expect(wrapper.state()).toEqual(expected);
  });

  it("should invoke handleSubmit onClick of submit button", () => {
    wrapper.instance().handleSubmit = jest.fn();
    wrapper.instance().forceUpdate();
    wrapper.find("#Form__btn--submit").simulate("click");
    expect(wrapper.instance().handleSubmit).toHaveBeenCalled();
  });

  // it('should invoke handleChange onChange of inputs', () => {
  //   wrapper.instance().handleChange = jest.fn();
  //   wrapper.instance().forceUpdate();
  //   wrapper.find("#Form__input--date").simulate("keydown")
  //   expect(wrapper.instance().handleChange).toHaveBeenCalled();
  // })
});
