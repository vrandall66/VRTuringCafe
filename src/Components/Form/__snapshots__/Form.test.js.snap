// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot 1`] = `
<form
  className="Form>"
>
  <input
    id="Form__input--name"
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    id="Form__input--date"
    name="date"
    onChange={[Function]}
    placeholder="Date (mm/dd)"
    type="text"
    value=""
  />
  <input
    id="Form__input--time"
    name="time"
    onChange={[Function]}
    placeholder="Time"
    type="text"
    value=""
  />
  <input
    id="Form__input--number"
    name="number"
    onChange={[Function]}
    placeholder="Number of Guests"
    type="text"
    value=""
  />
  <button
    id="Form__btn--submit"
    name="submit"
    onClick={[Function]}
    type="button"
  >
    Make Reservation
  </button>
</form>
`;
